# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1451047079 18000
#      Fri Dec 25 07:37:59 2015 -0500
# Node ID 167c536419a7def253d114d7d625d2dca83d864e
# Parent  004fa31be0b77636aed5d94974d595e01b3bde4f
deprecate C++ usage function

* error.h (usage, vusage): Tag with GCC_ATTR_DEPRECATED.

diff --git a/libinterp/corefcn/error.h b/libinterp/corefcn/error.h
--- a/libinterp/corefcn/error.h
+++ b/libinterp/corefcn/error.h
@@ -38,18 +38,18 @@ extern OCTINTERP_API void reset_error_ha
 
 extern OCTINTERP_API int warning_enabled (const std::string& id);
 
 extern OCTINTERP_API void
 vmessage (const char *name, const char *fmt, va_list args);
 
 extern OCTINTERP_API void message (const char *name, const char *fmt, ...);
 
-extern OCTINTERP_API void vusage (const char *fmt, va_list args) GCC_ATTR_NORETURN;
-extern OCTINTERP_API void usage (const char *fmt, ...) GCC_ATTR_NORETURN;
+extern OCTINTERP_API void vusage (const char *fmt, va_list args) GCC_ATTR_NORETURN GCC_ATTR_DEPRECATED;
+extern OCTINTERP_API void usage (const char *fmt, ...) GCC_ATTR_NORETURN GCC_ATTR_DEPRECATED;
 
 extern OCTINTERP_API void vwarning (const char *fmt, va_list args);
 extern OCTINTERP_API void warning (const char *fmt, ...);
 
 extern OCTINTERP_API void verror (const char *fmt, va_list args) GCC_ATTR_NORETURN;
 extern OCTINTERP_API void error (const char *fmt, ...) GCC_ATTR_NORETURN;
 
 extern OCTINTERP_API void verror (octave_execution_exception&,
